gem2rpm/foreman_plugin.spec.erb

```diff
@@ -0,0 +1,184 @@
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel <%= req %>
+<% end -%>
+<% if spec.extensions.empty? -%>
+BuildArch: noarch
+<% end -%>
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+Provides: foreman-plugin-%{plugin_name}
+
+%description
+<%= spec.description %>
+
+<% if doc_subpackage -%>
+%package doc
+Summary: Documentation for %{pkg_name}
+Group: Documentation
+Requires: %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+BuildArch: noarch
+
+%description doc
+Documentation for %{pkg_name}.
+<% end # if doc_subpackage -%>
+
+%prep
+%{?scl:scl enable %{scl} - << \EOF}
+gem unpack %{SOURCE0}
+%{?scl:EOF}
+
+%setup -q -D -T -n  %{gem_name}-%{version}
+
+%{?scl:scl enable %{scl} - << \EOF}
+gem spec %{SOURCE0} -l --ruby > %{gem_name}.gemspec
+%{?scl:EOF}
+
+%build
+# Create the gem as gem install only works on a gem file
+%{?scl:scl enable %{scl} - << \EOF}
+gem build %{gem_name}.gemspec
+%{?scl:EOF}
+
+# %%gem_install compiles any C extensions and installs the gem into ./%%gem_dir
+# by default, so that we can move it into the buildroot in %%install
+%{?scl:scl enable %{scl} - << \EOF}
+%gem_install
+%{?scl:EOF}
+
+%install
+mkdir -p %{buildroot}%{gem_dir}
+cp -pa .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
+
+<% unless spec.extensions.empty? -%>
+mkdir -p %{buildroot}%{gem_extdir_mri}
+cp -a .%{gem_extdir_mri}/{gem.build_complete,*.so} %{buildroot}%{gem_extdir_mri}/
+
+<% for ext in spec.extensions -%>
+# Prevent dangling symlink in -debuginfo (rhbz#878863).
+rm -rf %{buildroot}%{gem_instdir}/<%= ext.split(File::SEPARATOR).first %>/
+<% end -%>
+
+<% end -%>
+<% unless spec.executables.nil? or spec.executables.empty? -%>
+mkdir -p %{buildroot}%{_bindir}
+cp -pa .%{_bindir}/* \
+        %{buildroot}%{_bindir}/
+find %{buildroot}%{gem_instdir}/bin -type f | xargs chmod a+x
+
+<% end -%>
+%foreman_bundlerd_file
+<% unless precompile_flags.strip.empty? -%>
+%foreman_precompile_plugin <%= precompile_flags.strip %>
+<% end -%>
+
+%files
+%dir %{gem_instdir}
+<% unless spec.executables.nil? or spec.executables.empty? -%>
+<% for f in spec.executables -%>
+%{_bindir}/<%= f %>
+<% end -%>
+<% end -%>
+<% unless spec.extensions.empty? -%>
+%exclude %{gem_instdir}/ext
+%{gem_extdir_mri}
+<% end -%>
+<%= main_file_entries(spec).gsub(/^%license/, '%doc') %>
+<% unless doc_subpackage -%>
+%doc %{gem_docdir}
+<%= doc_file_entries(spec) %>
+<% end -%>
+%exclude %{gem_cache}
+%{gem_spec}
+%{foreman_bundlerd_plugin}
+<% if has_apidoc -%>
+%{foreman_apipie_cache_foreman}
+%{foreman_apipie_cache_plugin}
+<% end -%>
+<% if has_assets -%>
+%{foreman_assets_plugin}
+<% end -%>
+
+<% if doc_subpackage -%>
+%files doc
+%doc %{gem_docdir}
+<%= doc_file_entries(spec) %>
+<% end # if doc_subpackage -%>
+
+%posttrans
+<% if has_db_migrations -%>
+%{foreman_db_migrate}
+<% end -%>
+<% if has_db_seeds -%>
+%{foreman_db_seed}
+<% end -%>
+<% if has_apidoc -%>
+%{foreman_apipie_cache}
+<% end -%>
+%{foreman_restart}
+exit 0
```

Comment:
%foreman_restart will exit with a non-zero code if the Foreman service isn't running or if it's running under Passenger, so we usually need to ignore it.

I do plan on improving the posttrans snippets so they support Passenger, and would probably fix the error handling at the same time.